### PR TITLE
style: Don't unconditionally extend() the declaration block vector.

### PR DESCRIPTION
**Pull request by emilio**
Since it appears in profiles when used from CSSOM, like the one in the bug
mentioned in the comment.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19530)

### CODE REVIEW

File: components/style/properties/declaration_block.rs [hunk 1]
```diff
@@ -399,15 +399,26 @@ impl PropertyDeclarationBlock {
...
+                    AllShorthand::CSSWideKeyword(_) |
+                    AllShorthand::WithVariables(_) => ShorthandId::All.longhands().len()
+                };
+                let push_calls_count =
+                    drain.declarations.len() + all_shorthand_len;
+
+                // With deduplication the actual length increase may be less than this.
+                self.declarations.reserve(push_calls_count);
+            }
+            DeclarationSource::CssOm => {
```
Comment (jdm) [inline components/style/properties/declaration_block.rs]: Would it make sense to keep the reserve when self.declarations is empty?

**Review by jdm — APPROVED** [review]

File: components/style/properties/declaration_block.rs (hunk 1 above)
Comment (emilio) [reply components/style/properties/declaration_block.rs]: Perhaps! Not sure it's common enough, but... I guess

Comment (emilio) [reply components/style/properties/declaration_block.rs]: Hmm, it's kind of annoying to duplicate the code to get `push_calls_count` and such.

**Comment by emilio** [comment]
@bors-servo r=jdm

 * I left a TODO because I don't think the code duplication is worth it.

**Comment by emilio** [comment]
>  * I left a TODO because I don't think the code duplication is worth it.

(Without measuring, I mean)

**Pull request merged**